pkg/configv2/default.go

```diff
@@ -0,0 +1,213 @@
+	},
+	DataDir:            "~/.bacalhau",
+	NameProvider:       "puuid",
+	StrictVersionMatch: false,
+	Orchestrator: types.Orchestrator{
+		Enabled: true,
+		Listen:  "0.0.0.0:4222",
+		Cluster: types.Cluster{
+			Listen: "0.0.0.0:4222",
+		},
+		NodeManager: types.NodeManager{
+			GCThreshold:       Day,
+			GCInterval:        10 * Minuet,
+			DisconnectTimeout: Minuet,
+			ManualApproval:    false,
+		},
+		StateStore: types.OrchestratorStateStore{
+			JobGCInterval:   10 * Minuet,
+			JobGCThreshold:  Day,
+			EvalGCThreshold: Hour,
+			Backend: types.StoreBackend{
+				Type:   "BoltDB",
+				Config: make(map[string]string),
+			},
+		},
+		Scheduler: types.Scheduler{
+			WorkerCount:          runtime.NumCPU(),
+			HousekeepingInterval: 30 * Second,
+			HousekeepingTimeout:  2 * Minuet,
+		},
+		EvaluationBroker: types.EvaluationBroker{
+			VisibilityTimeout: Minuet,
+			MaxRetryCount:     10,
+		},
+	},
+	Compute: types.Compute{
+		Enabled: true,
+		Heartbeat: types.Heartbeat{
+			InfoUpdateInterval:     Minuet,
+			ResourceUpdateInterval: 30 * Second,
+			Interval:               15 * Second,
+		},
+		TotalCapacity: types.Resource{
+			CPU:    "1",
+			Memory: "1Gb",
+			Disk:   "1Gb",
+			GPU:    "0",
+		},
+		AllocatedCapacity: types.ResourceScaler{
+			CPU:    "80%",
+			Memory: "80%",
+			Disk:   "80%",
+			GPU:    "100%",
+		},
+		StateStore: types.ComputeStateStore{
+			ExecutionGCInterval:  10 * Minuet,
+			ExecutionGCThreshold: Day,
+			Backend: types.StoreBackend{
+				Type:   "BoltDB",
+				Config: make(map[string]string),
+			},
+		},
+	},
+	WebUI: types.WebUI{
+		Enabled: false,
+		Listen:  "0.0.0.0:8483",
+	},
+	InputSources: types.InputSourcesConfig{
+		ReadTimeout: 5 * Minuet,
+		MazSize:     "100GB",
+		Decompression: types.DecompressionConfig{
+			SizeLimit:      "100GB",
+			FileCountLimit: 1000,
+		},
+		Config: map[string]map[string]interface{}{
+			models.StorageSourceURL: make(map[string]interface{}),
+			// TODO(forrest) [review]: we need this here to support running wasm jobs
+			// as the wasm binary is inlined
+			models.StorageSourceInline: make(map[string]interface{}),
+		},
+	},
+	Publishers: types.PublishersConfig{
+		Config: map[string]map[string]interface{}{
+			models.PublisherLocal: {
+				// TODO(forrest) [review]: this was the default, but public isn't a valid host. It could be worth changing.
+				"address": "public",
+				"port":    6001,
+			},
+		},
+	},
+	Executors: types.ExecutorsConfig{
+		Config: map[string]map[string]interface{}{
+			models.EngineDocker: {
+				"manifestcache": map[string]interface{}{
+					"size":             1000,
+					"duration":         Hour,
+					"refreshFrequency": Hour,
+				},
+			},
+			models.EngineWasm: make(map[string]interface{}),
+		},
+	},
+	ResultDownloaders: types.ResultDownloaders{
+		Timeout: 5 * Minuet,
+		Config: map[string]map[string]string{
+			models.StorageSourceS3:  make(map[string]string),
+			models.StorageSourceURL: make(map[string]string),
+		},
```

Comment:
Same as above. We want the engines to be auto-enabled and doesn't require the user's intervention to enable each in the config. The preference is opt-out behaviour by allowing the user to list the disabled engines instead. My preference is to have a list of disabled downloaders, and in the future we can also add a list of enabled downlaoders. If both are empty, then all are enabled. If disabled is defined, then all are enabled except the disabled. If enabled is not empty, then only the selected ones are enabled

Feel free to add enabled list today if you prefere. Maybe `AllowListed` and `BlockListed` can be alternative names for enabled and disabled if you prefer

Reply:
Is the feedback here to remove these from the Config field definition too? Since they are all assumed to be enabled by default then there isn't any reason to include them in the Default config?

With respect to executors, the docker engine requires some configuration for its cache, should the defaults for that live in the docker executors engine code such that they are overridden when a config value is provided?

Reply:
the feedback is to remove these from the config. As to where to put default docker configs, I'll leave that to you. Both options sounds reasonable